app/services/ui_notifications.rb

```diff
@@ -0,0 +1,37 @@
+# Namespace for Foreman UI notifications event handling
+module UINotifications
+  class Base
+    attr_reader :subject, :event
+
+    def self.deliver!(subject, event)
+      new(subject, event).deliver!
+    end
+
+    def initialize(subject, event)
+      raise(Foreman::Exception, 'must provide notification subject') if subject.nil?
+      raise(Foreman::Exception, 'invalid event') unless allowed_events.include?(event)
```

Comment:
Missing translation, also I think it'd be useful to include the event that was passed in the text. `invalid_event %s` or something